src/MyList.java

```diff
@@ -0,0 +1,79 @@
+import java.util.AbstractList;
```

Comment:
MyList and tests are in the same package - seems encapsulation is broken

Reply:
Fixed it.